dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -158,40 +161,37 @@ public PagedDoraWorker(
     mUfs = UnderFileSystem.Factory.create(
         mRootUFS,
         UnderFileSystemConfiguration.defaults(Configuration.global()));
-    mUfsStatusCache = CacheBuilder.newBuilder()
-        .maximumSize(Configuration.getInt(PropertyKey.DORA_UFS_FILE_STATUS_CACHE_SIZE))
-        .expireAfterWrite(Configuration.getDuration(PropertyKey.DORA_UFS_FILE_STATUS_CACHE_TTL))
-        .build(new CacheLoader<String, DoraMeta.FileStatus>() {
-          @Override
-          public DoraMeta.FileStatus load(String path) throws IOException {
-            UfsStatus status = mUfs.getStatus(path);
-            DoraMeta.FileStatus fs = buildFileStatusFromUfsStatus(status, path);
-            return fs;
-          }
-        });
     mListStatusCache = CacheBuilder.newBuilder()
         .maximumSize(Configuration.getInt(PropertyKey.DORA_UFS_LIST_STATUS_CACHE_NR_DIRS))
         .expireAfterWrite(Configuration.getDuration(PropertyKey.DORA_UFS_LIST_STATUS_CACHE_TTL))
         .build();
 
     mPageSize = Configuration.global().getBytes(PropertyKey.WORKER_PAGE_STORE_PAGE_SIZE);
+    // TODO(elega) use dora specific configurations
     mBlockMasterClientPool = new BlockMasterClientPool();
+    LockPool<String> pathLocks =
+        new LockPool<>((key) -> new ReentrantReadWriteLock(),
+            Configuration.getInt(PropertyKey.MASTER_LOCK_POOL_INITSIZE),
+            Configuration.getInt(PropertyKey.MASTER_LOCK_POOL_LOW_WATERMARK),
+            Configuration.getInt(PropertyKey.MASTER_LOCK_POOL_HIGH_WATERMARK),
+            Configuration.getInt(PropertyKey.MASTER_LOCK_POOL_CONCURRENCY_LEVEL));
 
     String dbDir = Configuration.getString(PropertyKey.DORA_WORKER_METASTORE_ROCKSDB_DIR);
     Duration duration = Configuration.getDuration(PropertyKey.DORA_WORKER_METASTORE_ROCKSDB_TTL);
     long ttl = duration.isNegative() ? -1 : duration.getSeconds();
     try {
       if (ttl == 0) {
         LOG.info("Worker MetaStore RocksDB TTL is configured to be ZERO. That means no cache!");
-        mMetaStore = null;
+        mMetaStore = new HeapDoraMetaStore();
       } else {
-        mMetaStore = new RocksDBDoraMetaStore(dbDir, ttl);
+        mMetaStore = new CachingDoraMetaStore(new RocksDBDoraMetaStore(dbDir, ttl), pathLocks);
       }
     } catch (RuntimeException e) {
       LOG.error("Cannot init RocksDBDoraMetaStore. Continue without MetaStore", e);
-      mMetaStore = null;
+      mMetaStore = new HeapDoraMetaStore();
     }
     mCacheManager = cacheManager;
+    mMetaManager = new DoraMetaManager(this, mMetaStore, mCacheManager, mUfs, pathLocks);
```

Comment:
so now the metstore can be either
1. HeapDoraMetaStore (heap only)
2. CachingdoraMetaStore (heap + rocks)

Reply:
How about
InMemoryMetaStore (heap only)
CachingPersistentMetaStore (heap + rocks)